### PR TITLE
Label styling and recognising labels-with-spaces

### PR DESCRIPTION
**Pull request by tsop14**
**Labels-with-spaces**
Filters labels-with-spaces (e.g. #NEW nauru pics from BD) so they become enclosed in quote marks and therefore searchable
Adds case to match phrases in imageSearch so that labels-with-spaces have .selected assigned to them
- label turns blue when selected
- label cannot be added twice

NOW:
labels

BEFORE:
labels2

**Styling**
Removes scroll bar under related labels
Moves semicolon to after the parent label

NOW:
screenshot-media local dev-gutools co uk 2015-10-12 16-24-58

BEFORE: 
screenshot-media local dev-gutools co uk 2015-10-12 16-17-18

### CODE REVIEW

File: kahuna/public/js/search/results.js [hunk 1]
```diff
@@ -207,7 +210,8 @@ results.controller('SearchResultsCtrl', [
 
         ctrl.setParentLabel = () => {
             if (ctrl.parentLabel) {
-                $state.transitionTo($state.current, { query: `#${ctrl.parentLabel}` }, {
+                const parentLabel = queryLabelFilterFilter(ctrl.parentLabel);
+                $state.transitionTo($state.current, { query: `${parentLabel}` }, {
```
Comment (theefer) [inline kahuna/public/js/search/results.js]: Note that ``` `${x}` ``` is a bit of a tautology for just `x`.

**Comment by theefer** [comment]
:+1: but this will conflict with @jamesgorrie's PR

File: kahuna/public/stylesheets/main.css [hunk 2]
```diff
@@ -974,7 +974,8 @@ input.related-labels__parent {
 }
 
 .related-labels__labels {
-    overflow: auto;
+    overflow-x: auto;
```
Comment (jamesgorrie) [inline kahuna/public/stylesheets/main.css]: This won't work due to the restrictions behind `overflow`. TL;DR; Anything with `overflow: auto` will have the other axis set to auto too.

File: kahuna/public/stylesheets/main.css [hunk 3]
```diff
@@ -957,7 +957,7 @@ input.search-query__input {
 .related-labels--with-labels {
     /* This is to allow scrolling of labels, but show the
     autocomplete dropdown */
-    overflow: auto;
+    overflow-x: auto;
```
Comment (jamesgorrie) [inline kahuna/public/stylesheets/main.css]: This will set both axis to auto AFAIK?

File: kahuna/public/stylesheets/main.css (hunk 2 above)
Comment (theefer) [inline kahuna/public/stylesheets/main.css]: Not sure what you mean, we tried with @tsop14 and it solved the problem?

Comment (jamesgorrie) [inline kahuna/public/stylesheets/main.css]: I'm getting it mixed up - it's when you mix it with `visible`, this doesn't wokr.
:+1:

**Comment by jamesgorrie** [comment]
> conflict

[I've added support in my branch](https://github.com/guardian/grid/commit/da5b57510e790128edd36b7cb6947a3c5a523ad9)

File: kahuna/public/stylesheets/main.css (hunk 3 above)
Comment (jamesgorrie) [inline kahuna/public/stylesheets/main.css]: Ignore - I'm being silly.

**Pull request merged**